pac4j-core/src/main/java/org/pac4j/core/client/IndirectClient.java

```diff
@@ -49,7 +49,7 @@ protected final void internalInit() {
         // check configuration
         CommonHelper.assertNotBlank("callbackUrl", this.callbackUrl, "set it up either on this IndirectClient or on the global Config");
         if (this.urlResolver == null) {
-            this.urlResolver = new DefaultUrlResolver();
+            this.urlResolver = new DefaultUrlResolver(true);
```

Comment:
I am not sure about this. This effectively changes the behavior of all IndirectClients who rely on defaults. Instead, why not do this for the SAML client only, or simply update the documentation to indicate how one would need to set the url-resolver if they need relative callback URLs?

Reply:
Yes, I hesitated before doing this change but had the same reasoning: why would a `UrlResolver` be instantiated without the resolution logic enabled?
There are actually three ways doing this:

1.  we leave it as it, and as you said, people will inject there one instance of `UrlResolver` with their appropriate settings - it's just one more things to do, things won't work out of the box
2. change the default `UrlResolver`in the `SamlClient`instead of the global `IndirectClient`, as mentionned by Misagh, making this breaking change more local
3. perform the global default in `IndirectClient` as pushed, which raises more questions about other clients and their behaviour

I tend to think that #2 is the best compromise, but if you want to keep with #1, that's all fine with me. Let me know.

Reply:
I'm more in favor of 3: if we have this logic and I don't think it really matches real-world use cases, let's use it, it will be sufficient for tests as least. Or option 4, let's remove it.

Reply:
OK - so I leave the change originally proposed in the PR (copied below as well for convenience), right ?
```
-            this.urlResolver = new DefaultUrlResolver();
+            this.urlResolver = new DefaultUrlResolver(true);
```

Reply:
No. You will need to revert the change back to original. Changing behavior globally for all clients/users is never ever acceptable. 

People will inject their one instance of UrlResolver with their appropriate settings. It's one more thing to do, and things do work out of the box except for those who have a different specialized environment than the 90% norm in which case we expect them to be capable enough and have the technical know-how to make a one-liner change in their configuration of the pac4j library.

Reply:
That said, it's a major version (v4) so we can change things globally. Does it make sense? I think it won't change the behavior for most people, but as we have this extra behavior available, I tend to agree...

Reply:
I don't necessarily agree. Yes, a major version allows one to theoretically change things, assuming there is an understanding of what major/minor means in semantic terms. However, just as a policy allows one to change things, it doesn't mean that we should. Breaking code or behavior, regardless of what some document says, is never a pleasant experience for upgrades and users.

Specially in this case, the change in question is a one-liner that can be changed by those interested with a specialized set-up. So it's not a huge effort or fix to expect. There is also no test case to ensure this change breaks existing behavior or covers new ground which makes it more difficult to verify. Finally, *invisible* behavioral changes are the worst when people upgrade. There is no warning that the behavior shall change and the only way to learn this is by shuffling through possible release notes. I am generally fine with breaking APIs for a new major release; Breaking behavior invisibly is not a good idea, again, for such as simple thing that anyone can simply adjust with one line.

Reply:
A compromise might be:

1. Activate this behavior conditionally, on-demand.
2. Warn users with a log message, indicating this behavior might at some point become the default.
3. Allow time for the message to sink in.
4. Then, make the change globally.

Reply:
Yes, indeed, but I'm not sure to see what can be: "conditionally, on-demand" in this case.

If we have a doubt, let's stick to the current behavior as it can still be enabled fairly easily.